federationapi/routing/join.go

```diff
@@ -64,7 +64,13 @@ func MakeJoin(
 	}
 
 	createJoinTemplate := func(proto *gomatrixserverlib.ProtoEvent) (gomatrixserverlib.PDU, []gomatrixserverlib.PDU, error) {
-		identity, signErr := cfg.Matrix.SigningIdentityFor(request.Destination())
+		// TODO: remove this once the join dance understands pseudo IDs
+		var dummyUserID *spec.UserID
+		dummyUserID, err = spec.NewUserID(fmt.Sprintf("@dummy:%s", request.Destination()), true)
```

Comment:
when does the db get updated with the actual userID after this point? wouldn't that be required for everything else to work?

Reply:
and wouldn't this possibly result in user overlap if multiple users from the same server join the room?
seems like a nightmare

Reply:
Looks like we don't even need to sign this event (we still do, though), so going to switch back to getting the signing identity for the `request.Destination()`